src/libcore/alloc.rs

```diff
@@ -161,9 +165,11 @@ impl Layout {
     /// `align` violates the conditions listed in
     /// [`Layout::from_size_align`](#method.from_size_align).
     #[unstable(feature = "alloc_layout_extra", issue = "55724")]
+    #[rustc_const_unstable(feature="const_alloc_layout", issue = "67521")]
     #[inline]
-    pub fn align_to(&self, align: usize) -> Result<Self, LayoutErr> {
-        Layout::from_size_align(self.size(), cmp::max(self.align(), align))
+    pub const fn align_to(&self, align: usize) -> Result<Self, LayoutErr> {
+        let align = if self.align() >= align { self.align() } else { align };
```

Comment:
Please make `cmp::max` `const fn`, too :)

Reply:
Should I file a separate tracking issue?

Reply:
sorry I missed this in the first pass

the PR lgtm after this

Reply:
Oh I just realized: `std::cmp::{max; min}` are defined as `pub fn max<T>(v1: T, v2: T) -> T where T: Ord`. There is no way to require an impl of `Ord` to be const?!

Reply:
Oh, no that's not possible. So leave `align_to` as not const fn for now, or is it super necessary?

Reply:
Not to me, no. I actually don't see this one-liner as a problem: The type is always going to be usize, so even if it never gets changed back to `cmp::max()`, it's totally obvious what it does.

Reply:
> it's totally obvious what it does.

I agree, but

* the line of what is obvious and what not is very subjective, if at some point we have to start arguing about it, it seems good to start out conservative if there's no strong desire for the main change
* the original author must have thought that `cmp::max` is even more obvious (and I think this, too). So making code less obvious just to make it const without a strong reason to do so seems not like a good strategy.

Thanks for changing it back

Reply:
Maybe a `const fn align_max(align1: usize, align2: usize) -> usize` helper function to replace `cmpp::max()`?